src/crimson/osd/object_context.h

```diff
@@ -130,11 +130,11 @@ class ObjectContext : public Blocker,
   void put_lock_type(RWState::State type) {
     switch (type) {
     case RWState::RWWRITE:
-      return rwlock.unlock_for_write();
+      return lock.unlock_for_write();
```

Comment:
note, the `get` and `put` usage is difficult to reason and debug. we might need to use RAII or a safer interface to expose this API. for instance, see https://github.com/scylladb/seastar/blob/master/include/seastar/core/rwlock.hh .

Reply:
@rzarzynski this is a follow-up of our discussion last night.

Reply:
https://trello.com/c/0PyCIhAL